src/plugins/kibana/server/routes/api/ingest/register_bulk.js

```diff
@@ -0,0 +1,98 @@
+import { Promise } from 'bluebird';
+import { parse, transform } from 'csv';
+import _ from 'lodash';
+import hi from 'highland';
+import { patternToIngest } from '../../../../common/lib/convert_pattern_and_ingest_name';
+import { PassThrough } from 'stream';
+import bulkRequestSchema from '../../../lib/schemas/bulk_request_schema';
+
+export function registerBulk(server) {
+  server.route({
+    path: '/api/kibana/{id}/_bulk',
+    method: 'POST',
+    config: {
+      payload: {
+        output: 'stream',
+        maxBytes: 1024 * 1024 * 1024
+      },
+      validate: {
+        payload: bulkRequestSchema
+      }
+    },
+    handler: function (req, reply) {
+      const boundCallWithRequest = _.partial(server.plugins.elasticsearch.callWithRequest, req);
+      const indexPattern = req.params.id;
+      const usePipeline = req.payload.pipeline;
+      const csv = req.payload.csv;
+      const fileName = csv.hapi.filename;
+      const responseStream = new PassThrough();
+      const parser = parse({
+        columns: true,
+        auto_parse: true,
+        delimiter: _.get(req.payload, 'delimiter', ',')
+      });
+
+      let currentLine = 2; // Starts at 2 since we parse the header separately
+
+      responseStream.write('[');
+      csv.pipe(parser);
+
+      hi(parser)
```

Comment:
> any developer at any time could make even trivial changes to this file that could break any or all of those things in subtle ways that could have effects we couldn't possibly anticipate through automated testing. That's why we should assert that these things are being setup/configured the way they need to be rather than hoping our tests will catch the unexpected effects of breaking them.

And I'd also like concrete examples of ways in which a developer could make trivial changes that break everything, where a unit test would save them, and they wouldn't just change the unit test in addition to the code if the subtle breakages are in fact impossible to anticipate and the developer is incapable of detecting them when testing their changes.